typescript/cli/src/deploy/warp.ts

```diff
@@ -313,3 +327,57 @@ async function getWarpCoreConfig(
 
   return warpCoreConfig;
 }
+
+export async function runWarpRouteApply(params: ApplyParams) {
+  const {
+    warpDeployConfig,
+    warpCoreConfig,
+    context: { registry, multiProvider },
+  } = params;
+
+  // Addresses used to get static Ism factories
+  const addresses = await registry.getAddresses();
+
+  // Convert warpCoreConfig.tokens[] into a mapping of { [chainName]: Config }
+  // This allows O(1) reads within the loop
+  const warpCoreByChain = Object.fromEntries(
+    warpCoreConfig.tokens.map((token) => [token.chainName, token]),
+  );
```

Comment:
This is fine but realistically if n < 1000-ish if really doesn't matter if it's O(1), O(n), or even O(n^2)

Reply:
True that

Reply:
I mostly did it because i like reasoning with similar data shapes. I thought it would be hard to compare an array with a keystore